File: Dart/lib/summaryRanges.dart
class Solution {
  List<String> summaryRanges(List<int> nums) {
    var ret = <String>[];
    if (nums.isEmpty) {
      return ret;
    }
    var start = nums[0];
    var end = nums[0];
    for (var i = 1; i < nums.length; i++) {
      if (nums[i] == end + 1) {
        end = nums[i];
      } else {
        if (start == end) {
          ret.add('$start');
        } else {
          ret.add('$start->$end');
        }
        start = nums[i];
        end = nums[i];
      }
    }
    if (start == end) {
      ret.add('$start');
    } else {
      ret.add('$start->$end');
    }
    return ret;
  }
}

File: Cs/Solution/SummaryRanges.cs

using System.Collections.Generic;

namespace Solution
{

	public partial class Solution
	{
		public IList<string> SummaryRanges(int[] nums)
		{
			var result = new List<string>();
			if (nums.Length == 0)
			{
				return result;
			}

			int start = nums[0];
			int end = start;
			for (int i = 1; i < nums.Length; i++)
			{
				if (nums[i] == end + 1)
				{
					end++;
					continue;
				}
				string range = (start == end) ? start.ToString() : $"{start}->{end}";
				result.Add(range);
				start = nums[i];
				end = start;
			}
			string lastRange = (start == end) ? start.ToString() : $"{start}->{end}";
			result.Add(lastRange);
			return result;
		}
	}
}
